### ISSUE TITLE
Worker shell break out

### ISSUE THREAD

**Issue by Mickael-van-der-Beek** [yaakov123/hagana#2]
Here's another little find.

It uses worker threads to spawn a new Node.js script for which Hagana wasn't loaded and executes a regular `child_process.execSync()` command.

lib-cmd.js:

```
import { Worker } from 'node:worker_threads';

export function run () {
  new Worker(
    './libs/lib-subcmd.js',
    {
      stderr: false,
      stdout: false
    }
  );
}
```

lib-subcmd.js:

```
import childProcess from 'child_process';

console.log('RESULT=', childProcess.execSync('echo "42"').toString());
```

**Comment by yaakov123** [comment]
Alright, this was a tough one. My solution is as follows:

1) Intercept calls to `new Worker()`
2) Resolve the file path provided to the worker constructor
3) Get the file contents from the resolved path
4) Determine if to inject Hagana using require or import and inject in the first line
5) Write the injected file contents to a tmp file somewhere
6) return a `new Worker(tmpFilePath)`

What do you say @Mickael-van-der-Beek

**Comment by yaakov123** [comment]
This is now fixed as of v0.0.14

**Comment by Mickael-van-der-Beek** [comment]
@yaakov123 Careful, your fix in `0.0.14` broke importing the `worker_thread` core module all together for me on Node.js v14:

```
internal/modules/cjs/loader.js:889
  const err = new Error(message);
              ^

Error: Cannot find module 'node:worker_threads'
Require stack:
- /my/folder/test-hagana/node_modules/@hagana/hagana/dist/hagana.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
    at Function.Module._load (internal/modules/cjs/loader.js:745:27)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (/my/folder/test-hagana/node_modules/@hagana/hagana/dist/hagana.js:22:2482)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at ModuleWrap.<anonymous> (internal/modules/esm/translators.js:199:29) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/my/folder/test-hagana/node_modules/@hagana/hagana/dist/hagana.js'
  ]
}
```
